src/main/java/teammates/storage/api/CoursesDb.java

```diff
@@ -148,7 +150,15 @@ private Course getCourseEntity(String courseId) {
     protected CourseAttributes makeAttributes(Course entity) {
         Assumption.assertNotNull(Const.StatusCodes.DBLEVEL_NULL_INPUT, entity);
 
-        return CourseAttributes.builder(entity.getUniqueId(), entity.getName(), entity.getTimeZone())
+        ZoneId courseTimeZone;
+        try {
+            courseTimeZone = ZoneId.of(entity.getTimeZone());
+        } catch (DateTimeException e) {
+            log.severe("Timezone '" + entity.getTimeZone() + "' of course '" + entity.getUniqueId()
+                    + "' is no longer supported. UTC will be used instead.");
```

Comment:
Change `no longer` to `not` so that the message is relevant for completely invalid data too.